packages/ra-ui-materialui/src/input/TextInput.spec.tsx

```diff
@@ -166,4 +166,36 @@ describe('<TextInput />', () => {
             expect(input.value).toEqual('bar');
         });
     });
+
+    describe('label', () => {
+        it('should not render label when `label` prop is `false`', () => {
```

Comment:
You're missing a test showing that a proper label prop renders a label.

Reply:
I've added two more tests: when `label` not specified and when `label` is a non-empty string